node/src/manager/commands/fix_block.rs

```diff
@@ -0,0 +1,304 @@
+                Some(diff) => {
+                    // Convert the JSON diff to a pretty-formatted text that will be displayed to
+                    // the user
+                    Some(diff_to_string(&diff, false))
+                }
+            }
+        }
+    }
+
+    /// Prints the difference between two [`serde_json::Value`] values to the user.
+    pub(super) fn report_difference(difference: Option<&str>, hash: &H256) {
+        if let Some(diff) = difference {
+            eprintln!("block {hash} diverges from cache:");
+            eprintln!("{diff}");
+        } else {
+            println!("Cached block is equal to the same block from provider.")
+        }
+    }
+
+    /// Attempts to delete a block from the block cache.
+    pub(super) fn delete_block(hash: &H256, chain_store: &ChainStore) -> anyhow::Result<()> {
+        println!("Deleting block {hash} from cache.");
+        chain_store.delete_blocks(&[&hash])?;
+        println!("Done.");
+        Ok(())
+    }
+
+    /// Queries the [`ChainStore`] about the chain head.
+    pub(super) fn find_chain_head(chain_store: &ChainStore) -> anyhow::Result<i32> {
+        let chain_head = chain_store.chain_head_ptr()?;
+        if let Some(block_ptr) = chain_head {
+            Ok(block_ptr.number)
+        } else {
+            anyhow::bail!("Could not find the chain head for {}", chain_store.chain)
+        }
+    }
+}
+
+mod helpers {
+    use super::*;
+    use graph::prelude::hex;
+    use std::io::{self, Write};
+
+    /// Tries to parse a [`H256`] from a hex string.
+    pub(super) fn parse_block_hash(hash: &str) -> anyhow::Result<H256> {
+        let hash = hash.trim_start_matches("0x");
+        let hash = hex::decode(hash)
+            .with_context(|| format!("Cannot parse H256 value from string `{}`", hash))?;
+        Ok(H256::from_slice(&hash))
+    }
+
+    /// Asks users if they are certain about truncating the whole block cache.
+    pub(super) fn prompt_for_confirmation() -> anyhow::Result<bool> {
+        print!("This will delete all cached blocks.\nProceed? [y/N] ");
+        io::stdout().flush()?;
+
+        let mut answer = String::new();
+        io::stdin().read_line(&mut answer)?;
+        answer.make_ascii_lowercase();
+
+        match answer.trim() {
+            "y" | "yes" => Ok(true),
+            _ => Ok(false),
+        }
+    }
+
+    /// Convenience function for extracting values from unary sets.
+    pub(super) fn get_single_item<I, T>(name: &'static str, collection: I) -> anyhow::Result<T>
+    where
+        I: IntoIterator<Item = T>,
+    {
+        let mut iterator = collection.into_iter();
+        match (iterator.next(), iterator.next()) {
+            (Some(a), None) => Ok(a),
+            (None, None) => bail!("Expected a single {name} but found none."),
+            _ => bail!("Expected a single {name} but found multiple occurrences."),
+        }
+    }
+}
+
+/// Custom range type that supports being parsed from a string.
+mod ranges {
```

Comment:
This module is an attempt to have ranges that can be parsed from strings like `"10..=15"` and `"20.."`, since Rust's built-in `Range` type is incapable of conversions.

Reply:
Removed in 180124537